### PR TITLE
Add a custom agent version label to the Windows Stackdriver Logging Agent logs.

### PR DESCRIPTION
**Pull request by iantalarico**
This will allow users to see which agent version they logged with and (although a little hacky) can see the version running on the machine.

### CODE REVIEW

File: windows-installer/setup.nsi [hunk 1]
```diff
@@ -220,6 +230,15 @@ Section "Install"
...
       StrCpy $2 "  pos_file '${MAIN_INSTDIR}\pos\winevtlog.pos'$\r$\n"
     ${EndIf}
+	
+    ; Count the number of 'VERSION_LABEL_PLACE_HOLDER' instances.  It should only
+    ; ever be 0 or 1.  We only have it in the template file once.
+    ${WordFind} "$3" "VERSION_LABEL_PLACE_HOLDER" "#" $4
+
+    ; If we hit the place holder line replace it with the proper label.
+    ${If} $4 == "1"
+      StrCpy $2 "    $\"agent$\":$\"windows-stackdriver-logging-agent ${VERSION}$\"$\r$\n"
```
Comment (igorpeshansky) [inline windows-installer/setup.nsi]: Do we really want to tag **every** log message with the agent version? Could we look at other mechanisms, e.g., write the version to the event log once on startup?

**Comment by iantalarico** [comment]
I think we do want to write this to every log.  We have no idea what all will be logged to to the instance and from where so I think it's good to differentiate this. 

For example we long the monitored resource or compute name on every log entry too.

**Comment by igorpeshansky** [comment]
@iantalarico Correct, but the monitored resource is required to identify where the log came from. Everything else is effectively baggage. I'll let @summit comment on this (Summit, please feel free to pull in others from the logging team).

**Comment by iantalarico** [comment]
Friendly Ping

**Comment by summit** [comment]
The agent version go in one of the user labels (logentry.labels field). The Monitored Resource labels are required and as Igor said identifies the source of the logs. Anything else is auxiliary information which increases your logs volume and counts against your bytes allotment. However, if users want the agent version, we can can probably add an option in the agent configuration to add the agent as one of the user labels.

**Comment by iantalarico** [comment]
@summit so just to make sure I understand correctly we should make this change an optional for users and/or should just log a single entry at start up?

**Comment by summit** [comment]
I think we can always write a single log entry at startup. However for long running agents, the first log entry will disappear after 7 or 30 days (depending on tier). So, we can also have an option to add a user label on each log entry (with default off).

**Comment by iantalarico** [comment]
alright thanks, I'll close this for now and circle back with a new fix

**Pull request closed without merging**